india_compliance/gst_india/doctype/purchase_reconciliation_tool/purchase_reconciliation_tool.js

```diff
@@ -991,3 +992,193 @@ function get_affected_rows(tab, selection, data) {
                     .length
         );
 }
+
+function set_value_color(wrapper, data) {
+    ic.reco_tool_detail_view_fields.forEach(field => {
+        if (!data.name || !data.isup_name) return;
+
+        if (!in_list(["place_of_supply", "is_reverse_charge"], field)) return;
+
+        if (data[field] != data[`isup_${field}`]) {
+            wrapper.find(`[data-label='${field}'], [data-label='isup_${field}']`).addClass('not-matched');
+        }
+    });
+}
+
+reco_tool.show_detailed_dialog = function (me, data) {
+    const mapped_data = get_mapped_invoice_data(data);
+
+    var d = new frappe.ui.Dialog({
+        title: "GSTR 2A / 2B vs Purchase Register",
+        fields: [
+            {
+                fieldtype: "HTML",
+                fieldname: "purchase_reco_tool_cards",
+            },
+            ...get_document_link_fields(me, mapped_data),
+            {
+                fieldtype: "HTML",
+                fieldname: "detail_view",
+            }
+        ],
+    });
+    _add_custom_actions(d, mapped_data);
+
+    const detail_view = d.fields_dict.detail_view.$wrapper;
+
+    // Render detail view dialog data
+    render_cards(mapped_data, d.fields_dict.purchase_reco_tool_cards.$wrapper);
+    detail_view.html(get_content_html(mapped_data));
+    set_value_color(detail_view, mapped_data);
+
+    d.show();
+};
+
+function get_mapped_invoice_data(data) {
+    let mapped_data = Object.assign({}, data);
+
+    ic.reco_tool_detail_view_fields.forEach(field => {
+        if (data.isup_name && !data.name) {
+            delete mapped_data[field];
+        } else if (data.name && !data.isup_name) {
+            delete mapped_data[`isup_${field}`];
+        }
+    });
+
+    mapped_data.supplier_gstin = mapped_data.supplier_gstin.replace(/\s+/g, '');
+    return mapped_data;
+}
```

Comment:
Data can be processed further to return only the fields that needs to be presented. This will reduce the HTML to a great extent.